doc/SONiC_local_users_password_reset_hld.md

```diff
@@ -0,0 +1,300 @@
+
+
+The ```reset_local_users_passwords_base.py``` will contain the following class:
+
+    ```
+    class LocalUsersConfigurationResetBase(object):
+        """
+        Base class for resetting local users' configuration on the switch
+        """
+        def should_trigger(self):
+            '''
+            define the condition to trigger
+            '''
+            # the condition to trigger start() method, the default implementation will be by checking if a long reboot press was detected.
+
+        def start(self):
+            '''
+            define the functionality
+            '''
+            # the implementation of deleting non-default users and restoring original passwords for default users and expiring them
+    ```
+
+
+Each vendor should inherent this class and add the implementation that he desires in this class.
+
+In the ```src/sonic-host-services/scripts/reset-local-users-passwords``` script, that it is being called on ```reset-local-users-passwords.service``` start, will do import of the inherented class and calls the ```start()``` method of the class as we can see in the following code snippet:
+
+
+    ```
+    def get_platform_local_users_passwords_reset():
+        try:
+            from sonic_platform.reset_local_users_passwords import LocalUsersConfigurationResetPlatform
+            local_users_password_reset_class = LocalUsersConfigurationResetPlatform()
+        except ImportError:
+            syslog.syslog(syslog.LOG_WARNING, "LocalUsersPasswordsReset: sonic_platform package not installed. Unable to find platform local users passwords reset implementation")
+            raise Exception('Local users passwords reset implementation is not defined')
+
+        return local_users_password_reset_class
+
+    ```
+
+
+The ```LocalUsersConfigurationResetPlatform``` class residing in ```platform/<vendor-path>/sonic_platform/reset_local_users_passwords.py``` that each vendor implements should look like this:
+
+
+    ```
+    class LongRebootPressPlatform(LongRebootPressBase):
+
+        def should_trigger(self):
+            '''
+            Vendor's specific condition to trigger
+            '''
+            # can inherent the base class implementation
+    
+        def start(self):
+            '''
+            Vendor specific implementation.
+            '''
+            # can inherent the base class implementation
+    ```
+
+
+The new serivce will call the python script ```src/sonic-host-services/scripts/reset-local-users-passwords``` which in return it will read the feature state from the ```LOCAL_USERS_PASSWORDS_RESET``` table in ```CONFIG_DB``` database. If feature is enabled and a long reset button press detected, the service will perfom the implementation in the ```start()``` in the vendor's specifc implementation as we can see in the following code snippet of ```src/sonic-host-services/scripts/reset-local-users-passwords```:
+
+
+```
+class LocalUsersConfigurationReset:
```

Comment:
I would recommend not to add code into the HLD. the HLD should contain design and flow diagrams. The code is subject to change and will lead to confusion in the future.